.github/workflows/changed_solutions.yml

```diff
@@ -0,0 +1,61 @@
+name: Changed Solutions
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    if: ${{ contains(github.actor, '[bot]') != true }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 2
+      - uses: actions/setup-node@v3
+        with:
+          node-version: "16"
+          cache: "npm"
+      - run: npm ci
+      - name: Get changed files
+        id: changed-files
+        uses: tj-actions/changed-files@v22
+      - name: Run solutions in changed files directory
+        run: |
```

Comment:
Similar to the other workflow, let's make a standalone `.sh` script for it. It'll be harder for people to run it individually because of the `${{ steps... }}` but I'm sure there's a good way to set that variable up locally (and if not, I'll want to eventually build it heh).

Reply:
I just checked for a git command. maybe we should use the git command instead of the changed file action. Can you check it out ? https://git-scm.com/docs/git-diff-tree . 

The command is `git diff-tree --dirstat=files --no-commit-id --name-only -r <branch_name>` , It brings out the list of changed files for the selected branch. so in the github action we can use `${{ github.head_ref}}` or if there is a better context to get the PR branch. 

Probably I just update the check-solutions.sh script to accept an argument (`$1`). and that way I can tweak the script to run all tests if root file changes, and specific project tests for project file changes.

Reply:
@JoshuaKGoldberg

Reply:
I'm dubious a Git command alone will work. What if someone sends a PR from a branch with several commits to a branch other than `main`? The action will need to have GitHub-specific information to know how far back in the history to search through.

Anyway my suggestion was just to cut+paste the code into a `.sh` script, not to significantly refactor. Unless I'm wrong and it's not possible to refer to the GitHub information in a shell script for some reason?